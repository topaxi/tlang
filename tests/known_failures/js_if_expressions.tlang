// Tests that fail in JavaScript backend due to if expression handling
// This demonstrates a limitation in the JavaScript code generation

fn test_return_if(times) {
    return if (times > 0) {
        "positive"
    } else {
        "zero or negative"
    };
}

fn countdown_recursive(times) {
    if times > 0 {
        times |> log();
        countdown_recursive(times - 1);
    }
}

log("=== JavaScript Backend Failures ===");

// These should work in interpreter but fail in JavaScript
test_return_if(3) |> log();
test_return_if(-1) |> log();

countdown_recursive(3);