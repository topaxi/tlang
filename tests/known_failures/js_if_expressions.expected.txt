=== JavaScript Backend Failures ===
positive
zero or negative
3
2
1